src/test/java/games/strategy/util/VersionTest.java

```diff
@@ -1,106 +1,61 @@
 package games.strategy.util;
 
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.lessThan;
 import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertFalse;
-import static org.junit.jupiter.api.Assertions.assertNotEquals;
-import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
 import org.junit.jupiter.api.Test;
 
-public class VersionTest {
+import nl.jqno.equalsverifier.EqualsVerifier;
 
+public class VersionTest {
   @Test
-  public void testCompare() {
-    final Version v1 = new Version(0, 0, 0);
-    final Version v2 = new Version(1, 0, 0);
-    assertNotEquals(v1, v2);
-    assertNotEquals(v2, v1);
+  public void shouldBeEquatableAndHashable() {
+    EqualsVerifier.forClass(Version.class).withIgnoredFields("exactVersion").verify();
   }
 
   @Test
-  public void testCompare2() {
-    final Version v1 = new Version(0, 0, 0);
-    final Version v2 = new Version(1, 1, 0);
-    assertNotEquals(v1, v2);
-    assertNotEquals(v2, v1);
-  }
+  public void testCompareTo() {
+    assertThrows(NullPointerException.class, () -> new Version(1, 0).compareTo(null));
```

Comment:
This test I think assumes an implementation detail, that the constructors are unified. If we were to refactor so this constructor were unique in null handling, but the others were then newly buggy for whatever reason, this test case would not catch the regression. To fix that we could add the additional `new Version` constructor variants to this test case, eg:
```
Arrays.stream(
  new Version(0,0),
  new Version(0,0,0),
  new Version(0,0,0,0))
.forEach(version ->  assertThrows(NullPointerException.class, () -> version.compareTo(null)));
```